great_tables/_formats.py

```diff
@@ -919,72 +1097,139 @@ def fmt_percent(
     else:
         scale_by = 1.0
 
-    # Generate a function that will operate on single `x` values in the table body
-    def fmt_percent_fn(
-        x: float,
-        decimals: int = decimals,
-        drop_trailing_zeros: bool = drop_trailing_zeros,
-        drop_trailing_dec_mark: bool = drop_trailing_dec_mark,
-        use_seps: bool = use_seps,
-        scale_by: float = scale_by,
-        sep_mark: str = sep_mark,
-        dec_mark: str = dec_mark,
-        force_sign: bool = force_sign,
-        placement: str = placement,
-        incl_space: bool = incl_space,
-    ):
-        # If the `x` value is a Pandas 'NA', then return the same value
-        if is_na(self._tbl_data, x):
-            return x
+    # TODO: fix type errors for members of `FormatFns`
+    return fmt(
+        self,
+        fns=FormatFns(
+            html=partial(
+                fmt_percent_context,
+                data=self,
+                decimals=decimals,
+                drop_trailing_zeros=drop_trailing_zeros,
+                drop_trailing_dec_mark=drop_trailing_dec_mark,
+                use_seps=use_seps,
+                scale_by=scale_by,
+                sep_mark=sep_mark,
+                dec_mark=dec_mark,
+                force_sign=force_sign,
+                placement=placement,
+                incl_space=incl_space,
+                pattern=pattern,
+                context="html",
+            ),
+            latex=partial(
+                fmt_percent_context,
+                data=self,
+                decimals=decimals,
+                drop_trailing_zeros=drop_trailing_zeros,
+                drop_trailing_dec_mark=drop_trailing_dec_mark,
+                use_seps=use_seps,
+                scale_by=scale_by,
+                sep_mark=sep_mark,
+                dec_mark=dec_mark,
+                force_sign=force_sign,
+                placement=placement,
+                incl_space=incl_space,
+                pattern=pattern,
+                context="latex",
+            ),
+            default=partial(
+                fmt_percent_context,
+                data=self,
+                decimals=decimals,
+                drop_trailing_zeros=drop_trailing_zeros,
+                drop_trailing_dec_mark=drop_trailing_dec_mark,
+                use_seps=use_seps,
+                scale_by=scale_by,
+                sep_mark=sep_mark,
+                dec_mark=dec_mark,
+                force_sign=force_sign,
+                placement=placement,
+                incl_space=incl_space,
+                pattern=pattern,
+                context="html",
+            ),
+        ),
+        columns=columns,
+        rows=rows,
+    )
 
-        # Scale `x` value by a defined `scale_by` value
-        x = x * scale_by
 
-        # Determine properties of the value
-        is_negative = _has_negative_value(value=x)
-        is_positive = _has_positive_value(value=x)
+def fmt_percent_context(
```

Comment:
The pattern continues, which is:

- extract internal formatting function to one that generalizes to different output contexts (naming it w/ suffix `_context`)
- replace `fmt()` statement in `fmt_*()` (e.g., `fmt_percent()`) with one that covers three contexts (`html`, `latex`, and `default` <- same as `html`) with `partial()`ed versions of the `fmt_*_context()` function